final_project_schema.sql

```diff
@@ -0,0 +1,190 @@
+CREATE DATABASE finalProject; --run me first!
+
+DROP DATABASE finalProject; --run me if you mess up
+
+DROP TABLE IF EXISTS users; --run the rest of the code now
+
+CREATE TABLE IF NOT EXISTS users (
+    username VARCHAR(50),
+    password VARCHAR(50) NOT NULL, --may need to change the length once we decide hashing length, for now store in plaintext
+    first_name VARCHAR(20) NOT NULL,
+    last_name VARCHAR(20) NOT NULL,
+    PRIMARY KEY (username)
+);
+
+DROP TABLE IF EXISTS vehicle;
+
+CREATE TABLE IF NOT EXISTS vehicle (
+    vehicle_id UUID,    -- this is just a basic vehicle
+    make VARCHAR(50) NOT NULL,
+    model VARCHAR(50) NOT NULL,
+    year CHAR(4) NOT NULL,
+    PRIMARY KEY (vehicle_id)
+);
+
+DROP TABLE IF EXISTS drive;
+
+CREATE TABLE IF NOT EXISTS drive (
+  drive_id UUID,
+    vehicle_id UUID NOT NULL, --assigns a vehicle to the drive
+    milage FLOAT NOT NULL,
+    duration INT NOT NULL,  --integer in minutes
+    caption VARCHAR(200) NOT NULL,
+    photo VARCHAR(500) NOT NULL, --link
+    date TIMESTAMP NOT NULL,    --timestamp for when the drive is added to database
+    username VARCHAR(50) NOT NULL, --assigns a drive to a user
+    PRIMARY KEY (drive_id),
+    FOREIGN KEY (vehicle_id) REFERENCES vehicle(vehicle_id),
+    FOREIGN KEY (username) REFERENCES users(username)
+);
+
+DROP TABLE IF EXISTS tags;
+
+CREATE TABLE IF NOT EXISTS tags (
+    drive_id UUID,
+    commute BOOL,   --for all of these, true = tagged, false or null = not tagged
+    near_death_experience BOOL,
+    carpool BOOL,
+    mostly_highway BOOL,
+    mostly_backroads BOOL,
+    PRIMARY KEY (drive_id)
+);
+
+DROP TABLE IF EXISTS likes;
+
+CREATE TABLE IF NOT EXISTS likes (
+  drive_id UUID, --the drive that user liked
+  username VARCHAR(50), --the user that liked it
+  PRIMARY KEY (drive_id,username),
+  FOREIGN KEY (drive_id) REFERENCES drive(drive_id),
+  FOREIGN KEY (username) REFERENCES users(username)
+);
+
+DROP TABLE IF EXISTS comments;
+
+CREATE TABLE IF NOT EXISTS comments (
+    drive_id UUID, --the drive that is commented
+    username VARCHAR(50), --the user that commented
+    comment VARCHAR(200) NOT NULL,
+    date TIMESTAMP NOT NULL, --when the comment was left
+    PRIMARY KEY (drive_id,username),
+    FOREIGN KEY (drive_id) REFERENCES drive(drive_id),
+    FOREIGN KEY (username) REFERENCES users(username)
+);
+
+DROP TABLE IF EXISTS user_vehicle;
+
+CREATE TABLE IF NOT EXISTS user_vehicle (
+    username VARCHAR(50), --attributes a vehicle to a user
+    vehicle_id UUID,
+    color VARCHAR(20) NOT NULL,
+    PRIMARY KEY (username, vehicle_id),
+    FOREIGN KEY (username) REFERENCES users(username),
+    FOREIGN KEY (vehicle_id) REFERENCES vehicle(vehicle_id)
+);
```

Comment:
I may be wrong, but is this table necessary? Each vehicle would belong to one user, why make a separate table?